xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -253,6 +253,19 @@ class PredicateSubtitlePriority
     return !filter(ss);
   }
 
+  bool IsNeedOpen(const SelectionStream& ss) const
+  {
+    if (relevant(ss))
+      return true;
+
+    // Even if the subtitle is "not relevant", it's necessary open it to read
+    // possible forced subtitles, only if the language is the same as the audio
+    if (g_LangCodeExpander.CompareISO639Codes(ss.language, audiolang))
+      return true;
```

Comment:
this is ambigue to understand
because this method can be called for all kind of use cases and not for forced only
and so applies to all

the comment is not so clear because you are talking of two different things
- `CDVDOverlay` with forced package (cue) metadata: force a stream to be shown, is called in a similar way "forced", as i can understand it can be a regular subtitle, and "forced" flag will not be shown in the GUI stream list as "forced"
- `SelectionStream` stream "forced": specifies the track as "forced", as we already know these are not regular subtitles, "forced" flag in this case will be shown in the GUI stream list 

since this is a particular use case that applies to VOBSUB/PGS only
it would be appropriate limit this behaviour for these codecs
and better describing a comment with reasons

the method name
`bool IsNeedOpen(const SelectionStream& ss) const`
it would make more sense something like "IsKeepStreamOpen"

Reply:
In current master code stream is keep open for all codecs (even is not necessary for e.g. SRT) so to keep current behavior and avoid possible unexpected bugs seems more safe maintain this code "agnostic to codec" and not change behavior or limit keep open only in PGS / VobSub.

This PR keep always open (same as before) except when language is different of audio and not any language selected by user in Player settings.

Reply:
at this point we might as well always close the stream
would have a more consistent behavior that dont depend on language code

> if (valid && !visible && !psp.KeepStreamOpen(ss))

to `if (valid && !visible)`

@enen92 an evaluation from you would be appreciated

Reply:
No!!!, this will broke all PGS forced subs inside the stream (like Avatar Blu-Ray, etc.)

Reply:
ah ok, since i don't have too time to dedicate these weeks i can't do extensive testing
anyway a look from @enen92 would be appropriate if he has time